llama_index/chat_engine/cohere_context_plus_citations.py

```diff
@@ -0,0 +1,170 @@
+import asyncio
+from threading import Thread
+from typing import Any, List, Optional, Tuple, Dict
+
+from llama_index.callbacks import trace_method
+from llama_index.chat_engine.types import (
+    AgentChatResponse,
+    StreamingAgentChatResponse,
+    ToolOutput,
+)
+from llama_index.core.base_retriever import BaseRetriever
+from llama_index.core.llms.types import ChatMessage, MessageRole
+from llama_index.llms.llm import LLM
+from llama_index.schema import NodeWithScore
+from llama_index.chat_engine import ContextChatEngine
+from llama_index.llms.cohere_utils import transform_nodes_to_cohere_documents_list
+
+
+class CohereContextPlusCitationsChatEngine(ContextChatEngine):
```

Comment:
We can discuss this further, but my feeling is that we can rename this to `CohereContextChatEngine`, this would imply that this class is more generic, and would allow it to be modified for future use (perhaps also when initialization the Chat class the kwargs can pass what fields to be returned?)